admin/src/components/generator/ContentGeneratorPanel.jsx

```diff
@@ -0,0 +1,482 @@
+import { useI18n } from '@wordpress/react-i18n';
+import { memo, useEffect, useRef, useState } from 'react';
+import '../../assets/styles/components/_ContentGeneratorPanel.scss';
+import { InputField, SingleSelectMenu, SuggestInputField } from '../../lib/tableImports';
+import Loader from '../Loader';
+import promptTemplates from '../../data/promptTemplates.json';
+import TextAreaEditable from '../../elements/TextAreaEditable';
+import EditableList from '../../elements/EditableList';
+import { postFetch } from '../../api/fetching';
+import Button from '../../elements/Button';
+import Checkbox from '../../elements/Checkbox';
+import {
+	augmentWithDomainContext,
+	augmentWithoutContext,
+	augmentWithURLContext, createPost,
+	getAugmentProcessResult, getPostTypes,
+} from '../../api/generatorApi';
+import { Editor as TinyMCE } from '@tinymce/tinymce-react/lib/cjs/main/ts/components/Editor';
+import { fetchLangs } from '../../api/fetchLangs';
+
+function ContentGeneratorPanel() {
+	const { __ } = useI18n();
+	const editorRef = useRef( null );
+	const [ editorVal, setEditorVal ] = useState( '' );
```

Comment:
huge mess

Reply:
As discussed, we're gonna lift the state up and use global storage for that. so all `useState` will be removed from this component